paper/example/ex_mnitst.jl

```diff
@@ -0,0 +1,345 @@
+    epoch_arr::Vector{Float64}
+    grads_arr::Vector{Float64} #TODO fix this type to be dynamic
+    ϵ::Float64 #TODO Fix with type T
+end
+
+# ####################################################
+# # different Layers used
+# ####################################################
+struct Conv
+    w::Any
+    b::Any
+    f::Any
+end
+(c::Conv)(x) = c.f.(pool(conv4(c.w, x) .+ c.b))
+Conv(w1, w2, cx, cy, f = relu) = Conv(param(w1, w2, cx, cy), param0(1, 1, cy, 1), f)
+
+struct Dense
+    w::Any
+    b::Any
+    f::Any
+    p::Any
+end
+# (d::Dense)(x) = d.f.(mat_zero_mean(d.w, mat(dropout(x, d.p))) .+ d.b)
+(d::Dense)(x) = d.f.(d.w * mat(dropout(x, d.p)) .+ d.b) # todo change * in future 
+
+Dense(i::Int, o::Int, f = sigm; pdrop = 0.0) = Dense(param(o, i), param0(o), f, pdrop)
+
+struct Chainnll <: KnetNLPModels.Chain
+    layers::Any
+    Chainnll(layers...) = new(layers)
+end
+(c::Chainnll)(x) = (for l in c.layers
+    x = l(x)
+end;
+x)
+(c::Chainnll)(x, y) = Knet.nll(c(x), y)  # nécessaire
+(c::Chainnll)(data::Tuple{T1,T2}) where {T1,T2} = c(first(data, 2)...)
+(c::Chainnll)(d::Knet.Data) = Knet.nll(c; data = d, average = true)
+
+
+"""
+Lenet model
+Data need to be MNIST (#todo test with CIFAR10))
+Returns Knet model and KnetNLPModel model
+"""
+function lenet_prob(xtrn, ytrn, xtst, ytst; minibatchSize = 100)
+
+    LeNet = Chainnll(
+        Conv(5, 5, 1, 20), # Conv(filter_length, filter_width, number of input channels from previous layer :n filters, 3 for RGB, 1 for Grey..., number of filters)
+        Conv(5, 5, 20, 50),
+        Dense(800, 500),
+        Dense(500, 10, identity),
+    )
+
+    LeNetNLPModel = KnetNLPModel(
+        LeNet;
+        data_train = (xtrn, ytrn),
+        data_test = (xtst, ytst),
+        size_minibatch = minibatchSize,
+    )
+
+
+    return LeNet, LeNetNLPModel
+
+end
+
+####################################################
+# R2 supports callback function so we don't need to 
+# init every epoch 
+# currently we have Cb stopping the DNN model after
+# some iterations, but TBD other stopping conditions
+####################################################
+
+function cb(
+    nlp,
+    solver,
+    stats,
+    data::StochasticR2Data,
+)
+    # Max epoch
+    if data.epoch == data.max_epoch
+        stats.status = :user
+        return
+    end
+
+    data.i = KnetNLPModels.minibatch_next_train!(nlp)
+    if data.i == 1   # once one epoch is finished     
+        # reset
+        data.grads_arr = []
+        data.epoch += 1
+        acc = KnetNLPModels.accuracy(nlp) # accracy of the minibatch on the test Data
+        train_acc = Knet.accuracy(nlp.chain; data = nlp.training_minibatch_iterator) #TODO minibatch acc.
+        @info @sprintf "Current epoch:  %d out of max epoch: %d, \t train_acc: %f \t test_acc: %f" data.epoch data.max_epoch train_acc acc
+        append!(data.train_acc_arr, train_acc) #TODO fix this to save the acc
+        append!(data.acc_arr, acc) #TODO fix this to save the acc
```

Comment:
Same comment, are `# ... ` still work in progess?

Reply:
I need to remove the TODO, they were for KnetNLPProblems to have function to do this, 
I will fix this shortly